src/falcon/falcon_c/rpo.h

```diff
@@ -0,0 +1,36 @@
+#include <stdint.h>
+#include <string.h>
+
+/* ==================================================================== */
+/*
+ * RPO hashing algorithm related structs and methods.
+ */
+
+typedef struct
+{
+    union
+    {
+        uint64_t A[12];
+        uint8_t dbuf[96];
+    } st;
+    uint64_t dptr;
+} inner_rpo128_context;
```

Comment:
It is using a Union so that we can view the state as both an array of `u8` and at the same time an array of `u64` (i.e., field elements).
I have added some explanatory words and removed the prefix.